plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/HiveTableOperations.java

```diff
@@ -296,6 +296,10 @@ private void refreshFromMetadataLocation(String newLocation)
                 .run(metadataLocation -> newMetadata.set(
                         TableMetadataParser.read(this, io().newInputFile(metadataLocation))));
 
+        if (newMetadata.get() == null) {
+            throw new TableNotFoundException(getSchemaTableName(), "Table metadata is missing.");
```

Comment:
`NOT_FOUND` is the wrong error code for this. We should do
```java
throw new TrinoException(ICEBERG_INVALID_METADATA, "Table metadata is missing or invalid: " + getSchemaTableName());
```
`IcebergMetadata.listTableColumns()` should ignore this error as well (the same way that `HiveMetadata` does)

Related, we should actually change the `Tasks` code above to do `throwFailureWhenFinished()` and wrapper this exception with the `TrinoException` so that we know happened (such as corrupt file).

Reply:
Thanks @electrum i will send a pr for comments below
```
Related, we should actually change the Tasks code above to do throwFailureWhenFinished() and wrapper this exception with the TrinoException so that we know happened (such as corrupt file). This could be a follow up to this PR, or feel free to combine it if you want.
```